qiskit/qobj/pulse_qobj.py

```diff
@@ -0,0 +1,610 @@
+
+        if kwargs:
+            self._data.update(kwargs)
+
+    @property
+    def meas_level(self):
+        """The measurement level to use."""
+        return self._data['meas_level']
+
+    @meas_level.setter
+    def meas_level(self, data):
+        self._data['meas_level'] = data
+
+    @property
+    def meas_return(self):
+        """The level of measurement information to return."""
+        return self._data['meas_return']
+
+    @meas_return.setter
+    def meas_return(self, data):
+        self._data['meas_return'] = data
+
+    @property
+    def shots(self):
+        """The number of shots."""
+        return self._data.get('shots')
+
+    @shots.setter
+    def shots(self, value):
+        self._data['shots'] = value
+
+    @property
+    def max_credits(self):
+        """The max_credits to use on the IBMQ public devices."""
+        return self._data.get('max_credits')
+
+    @max_credits.setter
+    def max_credits(self, value):
+        self._data['max_credits'] = value
+
+    @property
+    def memory_slots(self):
+        """The number of memory slots on the device."""
+        mem_slots = self._data.get('memory_slots')
+        if mem_slots is None:
+            raise AttributeError('Attribute memory_slots is not defined')
+        return mem_slots
+
+    @memory_slots.setter
+    def memory_slots(self, value):
+        self._data['memory_slots'] = value
+
+    def to_dict(self):
+        """Return a dictionary format representation of the Pulse Qobj config.
+
+        Returns:
+            dict: The dictionary form of the PulseQobjConfig.
+        """
+        out_dict = copy.copy(self._data)
+        if hasattr(self, 'pulse_library'):
+            out_dict['pulse_library'] = [
+                x.to_dict() for x in self.pulse_library]
+
+        return out_dict
+
+    @classmethod
+    def from_dict(cls, data):
+        """Create a new PulseQobjConfig object from a dictionary.
+
+        Args:
+            data (dict): A dictionary for the config
+
+        Returns:
+            PulseQobjConfig: The object from the input dictionary.
+        """
+        if 'pulse_library' in data:
+            pulse_lib = data.pop('pulse_library')
+            pulse_lib_obj = [PulseLibraryItem.from_dict(x) for x in pulse_lib]
+            data['pulse_library'] = pulse_lib_obj
+        return cls(**data)
+
+    def __getattr__(self, name):
+        try:
+            return self._data[name]
+        except KeyError:
+            raise AttributeError('Attribute %s is not defined' % name)
+
+    def __eq__(self, other):
+        if isinstance(other, PulseQobjConfig):
+            if self.to_dict() == other.to_dict():
+                return True
+        return False
+
+
+class PulseQobjExperiment:
+    """A Pulse Qobj Experiment.
+
+    Each instance of this class is used to represent an individual Pulse
+    experiment as part of a larger Pulse Qobj.
+    """
+
+    def __init__(self, config=None, header=None, instructions=None):
```

Comment:
Instructions is a required parameter.

Reply:
Fixed in: https://github.com/Qiskit/qiskit-terra/pull/3383/commits/05590e348cfe9a2e546d9fd76894f277d5f7edfc